### ISSUE TITLE
Render result difference from js (bug)

### ISSUE THREAD

**Issue by fe3dback** [zordius/lightncandy#245]
The PHP Code:
-------------

```php
require('./vendor/autoload.php');
use LightnCandy\LightnCandy;

// The Template:
$t = "
test {{name}} -- 
{{#each c}}
  {{#with this}}
    {{a}}={{../../name}}
  {{/with}}
{{/each}}
";

// Helpers:
$helpers = array();

// Partials:
$partials = array();

$phpStr = LightnCandy::compile($template, array(
  // Used compile flags
  'flags' => LightnCandy::FLAG_HANDLEBARSJS,
  'helpers' => $helpers,
  'partials' => $partials,
));

echo "Generated PHP Code:\n$phpStr\n";

// Input Data:
$data = [
  name => "world",
  c: [
    ["a"=>1],
    ["a"=>2],
    ["a"=>3]
  ]
);

// Save the compiled PHP code into a php file
file_put_contents('render.php', '<?php ' . $phpStr . '?>');

// Get the render function from the php file
$renderer = include('render.php');

echo "Result:\n" . $renderer($data);
```

The Issue:
----------

PHP Render this:
`test world -- 1=world2=world3=world`

JS (same template, same data) Render this:
`test world -- 1=2=3=`

JS (change template to):
```
test {{name}} -- 
{{#each c}}
  {{#with this}}
    {{a}}={{../name}} <-- removed here second "../"
  {{/with}}
{{/each}}
```
render as expected:
`test world -- 1=world2=world3=world`

**Comment by zordius** [comment]
Thank you for the report, will fix this later.

**Comment by zordius** [comment]
JSfiddle created: https://jsfiddle.net/jo7o2ygp/

**Comment by zordius** [comment]
fixed: `{{#with .}}` will not change context stack, this fix is in current master and 1.0.0-dev now, please try it, thanks.

**Comment by zordius** [comment]
the fixed is released in 1.0.0 now
